File: 1015 Smallest Integer Divisible by K.c


int smallestRepunitDivByK(int k){
    if (k % 2 == 0 || k % 5 == 0) {
        return -1;
    }
    int N = 0;
    for (int i = 1; i <= k; i++) {
        N = (N*10 + 1) % k;
        if (N == 0) {
            return i;
        }
    }
    return -1;
}

File: 1015 Smallest Integer Divisible by K.go
func smallestRepunitDivByK(k int) int {
    if (k % 2 == 0 || k % 5 == 0) {
        return -1
    }
    cur:= 0
    res:= 1
    for {
        cur = (cur * 10 + 1) % k
        if (cur == 0) {
            return res 
        }
        res += 1
    }
}
